streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -17,242 +17,225 @@
 
 import java.io.IOException;
-import java.io.UnsupportedEncodingException;
-import java.util.ArrayList;
-import java.util.Collections;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Map;
 import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
-import java.util.Properties;
+import java.util.Arrays;
+
 
 public class InternalTopicManager {
 
     private static final Logger log = LoggerFactory.getLogger(InternalTopicManager.class);
 
-    // TODO: the following ZK dependency should be removed after KIP-4
-    private static final String ZK_TOPIC_PATH = "/brokers/topics";
-    private static final String ZK_BROKER_PATH = "/brokers/ids";
-    private static final String ZK_DELETE_TOPIC_PATH = "/admin/delete_topics";
-    private static final String ZK_ENTITY_CONFIG_PATH = "/config/topics";
-    // TODO: the following LogConfig dependency should be removed after KIP-4
-    private static final String CLEANUP_POLICY_PROP = "cleanup.policy";
-    private static final String COMPACT = "compact";
-
-    private final ZkClient zkClient;
     private final int replicationFactor;
+    private final KafkaClient kafkaClient;
+    private StreamsKafkaClient streamsKafkaClient;
+    StreamsConfig config;
 
-    private class ZKStringSerializer implements ZkSerializer {
-
-        /**
-         * @throws AssertionError if the byte String encoding type is not supported
-         */
-        @Override
-        public byte[] serialize(Object data) {
-            try {
-                return ((String) data).getBytes("UTF-8");
-            } catch (UnsupportedEncodingException e) {
-                throw new AssertionError(e);
-            }
-        }
 
-        /**
-         * @throws AssertionError if the byte String encoding type is not supported
-         */
-        @Override
-        public Object deserialize(byte[] bytes) {
-            try {
-                if (bytes == null)
-                    return null;
-                else
-                    return new String(bytes, "UTF-8");
-            } catch (UnsupportedEncodingException e) {
-                throw new AssertionError(e);
-            }
-        }
-    }
+    private static final String CLEANUP_POLICY_PROP = "log.cleanup.policy";
+    private static final String COMPACT = "compact";
 
     public InternalTopicManager() {
-        this.zkClient = null;
+        this.config = null;
+        this.kafkaClient = null;
         this.replicationFactor = 0;
     }
 
-    public InternalTopicManager(String zkConnect, int replicationFactor) {
-        this.zkClient = new ZkClient(zkConnect, 30 * 1000, 30 * 1000, new ZKStringSerializer());
-        this.replicationFactor = replicationFactor;
+    public InternalTopicManager(StreamsConfig config) {
+        this.config = config;
+        this.streamsKafkaClient = new DefaultKafkaClientSupplier().getStreamKafkaClient(config);
+        this.kafkaClient = streamsKafkaClient.getKafkaClient();
```

Comment:
Rather than getting the `kafkaClient` why not push the functionality down to `StreamsKafkaClient` ? Hide the details and make it useful such that the internals aren't exposed?